tests/Doctrine/Tests/DBAL/DBALExceptionTest.php

```diff
@@ -21,4 +21,20 @@ public function testAvoidOverWrappingOnDriverException()
         $e = DBALException::driverExceptionDuringQuery($driver, $ex, '');
         $this->assertSame($ex, $e);
     }
+
+    public function testDriverRequiredWithUrl()
+    {
+        $url = 'mysql://localhost';
+        $exception = DBALException::driverRequired($url);
+
+        $this->assertInstanceOf('Doctrine\DBAL\DBALException', $exception);
+        $this->assertSame(
+            sprintf(
+                "The options 'driver' or 'driverClass' are mandatory if a connection URL without scheme " .
+                "is given to DriverManager::getConnection(). Given URL: %s",
+                $url
+            ),
```

Comment:
This could be just a string. Test should contain as less logic as possible :)

    "The options 'driver' or 'driverClass' are mandatory if a connection URL without scheme is given to DriverManager::getConnection(). Given URL: mysql://localhost"